core/sender_cacher.go

```diff
@@ -18,12 +18,28 @@ package core
 
 import (
 	"runtime"
+	"sync"
 
 	"github.com/ethereum/go-ethereum/core/types"
 )
 
-// SenderCacher is a concurrent transaction sender recoverer and cacher.
-var SenderCacher = newTxSenderCacher(runtime.NumCPU())
+var (
+	// senderCacherOnce is used to ensure that the SenderCacher is initialized only once.
+	// It leverages sync.Once to provide thread-safe lazy initialization.
+	senderCacherOnce sync.Once
+	// senderCacherInstance is a concurrent transaction sender recoverer and cacher.
+	senderCacherInstance = newTxSenderCacher(runtime.NumCPU())
```

Comment:
```suggestion
	senderCacherInstance *txSenderCacher
```
This is what you meant to do, right?

Reply:
yes, correct.